pkg/kubelet/allocation/allocation_manager_test.go

```diff
@@ -126,16 +126,45 @@ func TestUpdatePodFromAllocation(t *testing.T) {
 		},
 	}
 
+	podWithPodLevelResources := pod.DeepCopy()
+	podWithPodLevelResources.Spec.Resources = &v1.ResourceRequirements{
+		Requests: v1.ResourceList{
+			v1.ResourceCPU:    *resource.NewMilliQuantity(1200, resource.DecimalSI),
+			v1.ResourceMemory: *resource.NewQuantity(1700, resource.DecimalSI),
+		},
+		Limits: v1.ResourceList{
+			v1.ResourceCPU:    *resource.NewMilliQuantity(2000, resource.DecimalSI),
+			v1.ResourceMemory: *resource.NewQuantity(2500, resource.DecimalSI),
+		},
+	}
+
+	podWithoutContainerResources := pod.DeepCopy()
+	podWithoutContainerResources.Spec.Containers[0].Resources = v1.ResourceRequirements{}
+	podWithoutContainerResources.Spec.InitContainers[0].Resources = v1.ResourceRequirements{}
```

Comment:
I think the pod has 2 regular containers and 2 init containers. Also, did you mean for this pod to have pod-level resources?